app/Http/Controllers/TrainingDash.php

```diff
@@ -567,67 +573,366 @@ public function otsCancel($id) {
-            default: return $position;
+            default:
+                return $position;
+        }
+    }
+
+    public function statistics(Request $request) {
+        $yearSel = $monthSel = null;
+        if (isset($request->date_select)) {
+            $datePart = explode(' ', $request->date_select); // Format: MM YYYY
+            $monthSel = $datePart[0];
+            $yearSel = $datePart[1];
+        }
+        $stats = $this->generateTrainingStats($yearSel, $monthSel, 'stats');
+        return view('dashboard.training.statistics')->with('stats', $stats);
+    }
+
+    private function generateTrainingStats($year, $month, $dataType) {
+        $retArr = [];
+        $retArr['dateSelect'] = ['month' => $month, 'year' => $year];
+        // Set date range
+        if ($year == null) {
+            $year = Carbon::now()->format('Y');
+            $month = Carbon::now()->format('m');
         }
+        $from = Carbon::createFromDate($year, $month, 1)->startOfMonth()->toDateString();
+        $to = Carbon::createFromDate($year, $month, 1)->endOfMonth()->toDateString();
+        $retArr['date'] = ['start_date' => $from, 'end_date' => $to];
+        $sessions = TrainingTicket::whereBetween('start_date', [$from, $to])->get();
+        if ($dataType == 'stats') {
+            $sessionsPrevious = TrainingTicket::whereBetween('start_date', [Carbon::createFromDate($from)->startOfMonth()->subMonths(1)->toDateString(), Carbon::createFromDate($to)->startOfMonth()->subMonths(1)->toDateString()]);
+        }
+        // Training sessions per month
+        $retArr['sessionsPerMonth'] = $sessions->count();
+        if ($dataType == 'stats') {
+            $retArr['sessionsCompletePerMonth'] = $sessions->where('type', 12)->count();
+            $retArr['sessionsPerMonthNoShow'] = $sessions->where('type', 10)->count();
+            $retArr['sessionsPreviousMonth'] = $sessionsPrevious->count();
+            $retArr['sessionsCompletePreviousMonth'] = $sessionsPrevious->where('type', 12)->count();
+        }
+        // Training sessions per month by type
+        if ($dataType == 'graph') {
+            $sessionsS1 = $sessions->where('position', '<', 105)->count();
+            $sessionsS2 = $sessions->where('position', '>', 105)->where('position', '<', 115)->count();
+            $sessionsS3 = $sessions->whereIn('position', [115, 116, 117, 118, 119, 123])->count();
+            $sessionsC1 = $sessions->whereIn('position', [120, 121])->count();
+            $sessionsOther = $sessions->whereIn('position', [122, 124, 125])->count();
+            $retArr['sessionsByType'] = ['S1' => $sessionsS1, 'S2' => $sessionsS2, 'S3' => $sessionsS3, 'C1' => $sessionsC1, 'Other' => $sessionsOther];
+        }
+        // Training sessions, type, and hours by instructor
+        $trainers = $trainerSessions = [];
+        $activeRoster = User::where('status', '1')->where('visitor', '0')->get();
+        $activeRoster = $activeRoster->unique('id');
```

Comment:
Probably not a bad idea to create a new variable. PHP lets you change types on the fly but it can lead to issues in the long run.

Reply:
This actually isn't a type change (I know... it looks like it is). $activeRoster remains a collection even after the unique filter is applied. It is just returning a collection that consists of only rows with unique 'id' fields (filters all duplicates)

Reply:
Ah I see that now. This may be a case where it may be clearer to daisy chain the "unique" onto the end after the "get". Just a suggestion - it's about the same either way.

Reply:
Yeah, that works. I'll do that